src/fsql/column_parser.py

```diff
@@ -66,8 +66,17 @@ def is_terminal_level(self) -> bool:
         raise NotImplementedError("abc")
 
     @classmethod
-    def from_str(cls, path_description: str):
-        """Example: col1/col2=v1/col3=[v4,v5,v6]/colFname"""
+    def from_str(cls, path_description: str, fname: str | None = None):
+        """Creates a parser by processing partitions from the given path.
+
+        Example of path description: "col1/col2=v1/col3=[v4,v5,v6]". If filename should be parsed as well,
+        provide either the name of the partition (like `fname="fname"`) or as query (`fname="myfile=f1.csv"`).
```

Comment:
this actually got me confused a bit. In the first example, you'll end up with column `fname` in the result, holding the fname. In the second case, you'll end up with `myfile` column in the result, and it will only contain `f1.csv` values. Maybe expanding this in this sense would help understand it? I know that in the ColumnParser it feels weird to talk about "columns in the result", but its a docstring so we can perhaps afford to be a bit end-to-end

regarding the name of the variable -- maybe `fname_col_name`, or `fname_as`? The point is that both of them are confusing enough on a first read that it forces the reader to check the docstring, which is what we want. And on subsequent reads, it is "ah ok that was this confusing thing, I know". We could even name it "tom_bombadil" to reach the same effect, but `fname_col_name` or `fname_as` are perhaps more intuitive (and I want to use `tom_bombadil` in the grammar anyway)

Reply:
hm but the filter is not captured in there, good point... so `fname_query`? That is legit even if you don't include the filter -- I essentially view this as `select fname as fname_query.split(=)[0] where fname = fname_query.split(=)[1]`.